azure_monitor/src/azure_monitor/auto_collection/dependency_metrics.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import threading
+import time
+
+import requests
+from opentelemetry.metrics import LabelSet, Meter
+from opentelemetry.sdk.metrics import Gauge
+
+dependency_map = dict()
+_dependency_lock = threading.Lock()
+ORIGINAL_REQUEST = requests.Session.request
+
+
+def dependency_patch(*args, **kwargs) -> None:
+    result = ORIGINAL_REQUEST(*args, **kwargs)
+    # We don't want multiple threads updating this at once
+    with _dependency_lock:
+        count = dependency_map.get("count", 0)
+        dependency_map["count"] = count + 1
+    return result
+
+
+class DependencyMetrics:
+    def __init__(self, meter: Meter, label_set: LabelSet):
+        self._meter = meter
+        self._label_set = label_set
+        # Patch requests
+        requests.Session.request = dependency_patch
+        dependency_rate_metric = self._meter.create_metric(
+            "\\ApplicationInsights\\Dependency Calls/Sec",
+            "Outgoing Requests per second",
+            "rps",
+            int,
+            Gauge,
```

Comment:
With OT's [removal of gauge](https://github.com/open-telemetry/opentelemetry-python/issues/403), we will have to refactor this after the beta release. Do you still want to merge with these changes?

Reply:
Yes, refactor PR to align with OT beta code would be super clear instead of having huge PRs like this one